nifi-nar-bundles/nifi-enrich-bundle/nifi-enrich-processors/src/main/java/org/apache/nifi/processors/enrich/QueryDNS.java

```diff
@@ -0,0 +1,272 @@
+            getLogger().warn("Resolver was initialized at onTrigger instead of onScheduled");
+
+        }
+
+        FlowFile flowFile = session.get();
+        if (flowFile == null) {
+            return;
+        }
+
+        final String queryType = context.getProperty(DNS_QUERY_TYPE).getValue();
+        final String queryInput = context.getProperty(QUERY_INPUT).evaluateAttributeExpressions(flowFile).getValue();
+        final String queryParser = context.getProperty(QUERY_PARSER).getValue();
+        final String queryRegex = context.getProperty(QUERY_PARSER_INPUT).getValue();
+
+        boolean found = false;
+        try {
+            Attributes results = doLookup(queryInput, queryType);
+            // NOERROR & NODATA seem to return empty Attributes handled bellow
+            // but defaulting to not found in any case
+            if (results.size() < 1) {
+                found = false;
+
+
+            } else {
+                int recordNumber = 0;
+                NamingEnumeration<?> dnsEntryIterator = results.get(queryType).getAll();
+                while (dnsEntryIterator.hasMoreElements()) {
+                    String dnsRecord = dnsEntryIterator.next().toString();
+                    // While NXDOMAIN is being generated by doLookup catch
+                    if (dnsRecord != "NXDOMAIN") {
+                        Map<String, String> parsedResults = parseResponse(recordNumber, dnsRecord, queryParser, queryRegex, "dns");
+                        flowFile = session.putAllAttributes(flowFile, parsedResults);
+                        found = true;
+
+
+                    } else {
+                        // Otherwise treat as not found
+                        found = false;
+
+
+                    }
+                    // Increase the counter and iterate over next record....
+                    recordNumber++;
+                }
+            }
+        } catch (NamingException e) {
+            getLogger().error("Something went wrong while processing the Flowfile. Please review your configuration", e);
+            throw new ProcessException("Unexpected NamingException while processing records");
+        }
+
+
+        // Finally prepare to send the data down the pipeline
+        if (found) {
+            // Sending the resulting flowfile (with attributes) to REL_FOUND
+            session.transfer(flowFile, REL_FOUND);
+        } else {
+            // NXDOMAIN received, accepting the fate but forwarding
+            // to REL_NOT_FOUND
+            session.transfer(flowFile, REL_NOT_FOUND);
+        }
+    }
+
+    @OnScheduled
+    public void onScheduled(ProcessContext context) {
+        try {
+            initializeResolver(context);
+        } catch (Exception e) {
+            context.yield();
+            throw new ProcessException("Failed to initialize the JNDI DNS resolver server", e);
+        }
+    }
+
+    protected void initializeResolver(final ProcessContext context ) {
+        if (initialized.get()) {
```

Comment:
@trixpan yes.
Let's say I first configure my processor without a list of DNS servers. Then I run it, the resolver will be considered as initialized. Then I stop the processor and I update the list of DNS servers, but when starting the processor, since it has been initialized, it won't get updated.
If you want to keep this check, then another option is to add an ``@OnUnscheduled`` annotated method to change the boolean 'initialized' to false when the processor is stopped.